### PR TITLE
PLNSRVCE-938: check task and pipeline image repo existence

### PR DESCRIPTION
**Pull request by tkdchen**
Signed-off-by: Chenxiong Qi <cqi@redhat.com>

### CODE REVIEW

File: hack/check-task-pipeline-bundle-repos.sh [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    task_name=$(yq e '.metadata.name' "$task_file")
+    found=$(locate_bundle_repo "task-$task_name")
+    if [ "$found" != "200" ]; then
+	echo "Missing task bundle repo: task-$task_name"
+	has_missing_repo=yes
+    fi
+done
+
+# pipelines
+for pl_name in $(oc kustomize pipelines/ | yq e '.metadata.name' | grep -v "^-\+$"); do
```
Comment (Michkov) [inline hack/check-task-pipeline-bundle-repos.sh:26]: without yq:
```
oc kustomize pipelines/ | oc apply --dry-run=client -f- -o jsonpath='{range .items[*]}{.metadata.name}{"\n"}{end}'
```

File: hack/check-task-pipeline-bundle-repos.sh [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+locate_bundle_repo() {
+    local -r repo_name=$1
+    curl -I -s -L -w "%{http_code}\n" -o /dev/null "https://quay.io/api/v1/repository/${QUAY_ORG}/${repo_name}"
+}
+
+has_missing_repo=
+
+# tasks
+for task_file in task/*/*/*.yaml; do
+    task_name=$(yq e '.metadata.name' "$task_file")
```
Comment (Michkov) [inline hack/check-task-pipeline-bundle-repos.sh:17]: ` oc apply -f $task_file -o jsonpath='{.metadata.name}' --dry-run=client`

File: .tekton/pull-request.yaml [hunk 3]
```diff
@@ -150,6 +150,20 @@ spec:
                 # Perform cleanup of resources created by gitops service
                 oc delete --ignore-not-found deployment --all -n $(params.e2e_test_namespace)
                 oc delete --ignore-not-found eventlisteners --all -n $(params.e2e_test_namespace)
+      - name: check-task-pipeline-repo-existence
+        taskSpec:
+          steps:
+            - name: fail-when-repo-is-missed
+              image: quay.io/redhat-appstudio/pull-request-builds:build-definitions-utils-{{revision}}
```
Comment (Michkov) [inline .tekton/pull-request.yaml:157]: can be replaced by registry.redhat.io/openshift4/ose-cli:v4.11 - in case that you remove yq command

**Review by Michkov — COMMENTED** [review]
I want to stop generating appstudio-utils image, proposed solution how to use ose-cli image only

File: hack/check-task-pipeline-bundle-repos.sh (hunk 1 above)
Comment (tkdchen) [reply hack/check-task-pipeline-bundle-repos.sh:26]: `oc apply` connects to a OpenShift instance even if using `--dry-run=client`:

```
unable to recognize "STDIN": Get "https://api.ci-ln-0wjh7rb-76ef8.origin-ci-int-aws.dev.rhcloud.com:6443/api?timeout=32s": dial tcp: lookup api.ci-ln-0wjh7rb-76ef8.origin-ci-int-aws.dev.rhcloud.com: no such host
unable to recognize "STDIN": Get "https://api.ci-ln-0wjh7rb-76ef8.origin-ci-int-aws.dev.rhcloud.com:6443/api?timeout=32s": dial tcp: lookup api.ci-ln-0wjh7rb-76ef8.origin-ci-int-aws.dev.rhcloud.com: no such host
unable to recognize "STDIN": Get "https://api.ci-ln-0wjh7rb-76ef8.origin-ci-int-aws.dev.rhcloud.com:6443/api?timeout=32s": dial tcp: lookup api.ci-ln-0wjh7rb-76ef8.origin-ci-int-aws.dev.rhcloud.com: no such host
unable to recognize "STDIN": Get "https://api.ci-ln-0wjh7rb-76ef8.origin-ci-int-aws.dev.rhcloud.com:6443/api?timeout=32s": dial tcp: lookup api.ci-ln-0wjh7rb-76ef8.origin-ci-int-aws.dev.rhcloud.com: no such host
```

Does `registry.redhat.io/openshift4/ose-cli:v4.11` image have default configuration for that?

File: .tekton/pull-request.yaml (hunk 3 above)
Comment (Michkov) [reply .tekton/pull-request.yaml:157]: use your redhat.com credentials

File: hack/check-task-pipeline-bundle-repos.sh (hunk 1 above)
Comment (tkdchen) [reply hack/check-task-pipeline-bundle-repos.sh:26]: Looks some prerequisition is required for running `oc apply`:

```
resource mapping not found for name: "docker-build" namespace: "" from "STDIN": no matches for kind "Pipeline" in version "tekton.dev/v1beta1"
ensure CRDs are installed first
resource mapping not found for name: "enterprise-contract" namespace: "" from "STDIN": no matches for kind "Pipeline" in version "tekton.dev/v1beta1"
ensure CRDs are installed first
resource mapping not found for name: "java-builder" namespace: "" from "STDIN": no matches for kind "Pipeline" in version "tekton.dev/v1beta1"
ensure CRDs are installed first
resource mapping not found for name: "nodejs-builder" namespace: "" from "STDIN": no matches for kind "Pipeline" in version "tekton.dev/v1beta1"
ensure CRDs are installed first
```

Comment (Michkov) [reply hack/check-task-pipeline-bundle-repos.sh:26]: yop you need to have Openshift pipelines installed

**Review by Michkov — APPROVED** [review]

**Pull request merged**